server/google_resources.py

```diff
@@ -75,13 +105,12 @@ def post(self):
 
     # Use the authorization server's response to fetch the OAuth 2.0 tokens.
     authorization_response = flask.request.json["url"]
-    print(flask.request.json)
     flow.fetch_token(authorization_response=authorization_response)
 
     # Store credentials in the session.
     # ACTION ITEM: In a production app, you likely want to save these
     #              credentials in a persistent database instead.
     credentials = flow.credentials
     creds_dict = credentials_to_dict(credentials)
-    save_credentials(creds_dict)
+    save_credentials(creds_dict, user_id)
```

Comment:
This causes the server to break because user_id is not defined. To use the  `@login_required` decorator I made, just add `client_id` as a parameter like so..

```
  @login_required
  def post(self, client_id):
```